### PR TITLE
docs: update prerender documentation

### PR DESCRIPTION
**Pull request by Barbapapazes**
### 🔗 Linked issue

related https://github.com/nuxt/content/issues/2203#issuecomment-1658047596

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I've updated the section about the Nitro pre-rendering since I see a lot of questions or issue about it in the `nuxt/content` repo.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.getting-started/10.deployment.md [hunk 1]
```diff
@@ -81,52 +81,61 @@ There are two ways to deploy a Nuxt application to any static hosting services:
...
     prerender: {
       routes: ['/user/1', '/user/2']
+      ignore: ['/dynamic']
     }
   }
 })
 ```
 
-When using this option with `nuxi build`, static payloads won't be generated by default at build time. For now, selective payload generation is under an experimental flag.
+You can combine this with the `crawLinks` option to pre-render a set of routes that the crawler can't discover like you're `/sitemap.xml` or `/robots.txt`:
```
Comment (danielroe) [inline docs/1.getting-started/10.deployment.md:120]:
```suggestion
You can combine this with the `crawLinks` option to pre-render a set of routes that the crawler can't discover like your `/sitemap.xml` or `/robots.txt`:
```

File: docs/1.getting-started/10.deployment.md [hunk 2]
```diff
@@ -81,52 +81,61 @@ There are two ways to deploy a Nuxt application to any static hosting services:
 
 ### Crawl-based Pre-rendering
 
-Use the [`nuxi generate` command](/docs/api/commands/generate) to build your application. For every page, Nuxt uses a crawler to generate a corresponding HTML and payload files. The built files will be generated in the `.output/public` directory.
+Use the [`nuxi generate` command](/docs/api/commands/generate) to build and pre-render your application using the [Nitro](/docs/guide/concepts/server-engine) crawler. This command is similar to `nuxt build` with the `nitro.prerender` option set to `true`.
```
Comment (danielroe) [inline docs/1.getting-started/10.deployment.md:84]:
```suggestion
Use the [`nuxi generate` command](/docs/api/commands/generate) to build and pre-render your application using the [Nitro](/docs/guide/concepts/server-engine) crawler. This command is similar to `nuxt build` with the `nitro.static` option set to `true`, or running `nuxt build --prerender`.
```

File: docs/1.getting-started/10.deployment.md [hunk 3]
```diff
@@ -81,52 +81,61 @@ There are two ways to deploy a Nuxt application to any static hosting services:
...
-  nitro: {
-    prerender: {
-      crawlLinks: true
-    }
-  }
-})
-```
+Working of the Nitro crawler:
+
+1. Load the HTML of your application's root route (`/`).
```
Comment (danielroe) [inline docs/1.getting-started/10.deployment.md:94]:
```suggestion
1. Load the HTML of your application's root route (`/`), any non-dynamic pages in your `~/pages` directory, and any other routes in the `nitro.prerender.routes` array.
```

File: docs/1.getting-started/10.deployment.md [hunk 4]
```diff
@@ -81,52 +81,61 @@ There are two ways to deploy a Nuxt application to any static hosting services:
...
+2. Save the HTML to a file in the `.output/public` directory for being served statically.
```
Comment (danielroe) [inline docs/1.getting-started/10.deployment.md:95]:
```suggestion
2. Save the HTML and `payload.json` to the `~/.output/public/` directory to be served statically.
```

File: docs/1.getting-started/10.deployment.md (hunk 2 above)
Comment (Barbapapazes) [reply docs/1.getting-started/10.deployment.md:84]: 🤔 There is no such option in the Nitro API https://nitro.unjs.io/config#static.

Will investigate in the Nitro documentation.

Comment (Barbapapazes) [reply docs/1.getting-started/10.deployment.md:84]: Ok, I see, that's map to a preset.

https://github.com/unjs/nitro/blob/main/src/presets/static.ts

**Review by danielroe — APPROVED** [review]

**Pull request merged**